addons/cloud_storage/models/ir_attachment.py

```diff
@@ -0,0 +1,88 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import uuid
+
+from odoo import models, fields, _
+from odoo.exceptions import UserError
+from odoo.http import Stream
+
+
+class CloudStorageAttachment(models.Model):
+    _inherit = 'ir.attachment'
+    _cloud_storage_upload_url_time_to_expiry = 300  # 300 seconds
+    _cloud_storage_download_url_time_to_expiry = 300  # 300 seconds
+
+    type = fields.Selection(
+        selection_add=[('cloud_storage', 'Cloud Storage')],
+        ondelete={'cloud_storage': 'set url'}
+    )
+
+    def _to_http_stream(self):
+        if (self.type == 'cloud_storage' and
+              self.env['res.config.settings']._get_cloud_storage_configuration()):
+            self.ensure_one()
+            stream = Stream()
```

Comment:
Shouldn't it be something like
```suggestion
            stream = Stream(
                mimetype=self.mimetype,
                download_name=self.name,
                conditional=True,
                etag=self.checksum,
            )
```

Reply:
checked with @Julien00859 It is OK, because it is a redirection, all these info are not needed and won't be used

Reply:
@HydrionBurst thanks for the explanation.

Reply:
+1 those 4 attributes are useless when `Stream.type == 'url'`